### PR TITLE
Added sample_names attribute and fixed test

### LINKED ISSUE #28: add sample names attribute
add sample names attribute in the ramclustr object and do the following

- #29
- #30

### LINKED ISSUE #29: use sample names attribute instead of phenoData
use sample names attribute to replace the use of sample names from phenoData

### LINKED ISSUE #30: Fix test cases after introducing sample_name attribute
fix all test cases which broke because of sample names attribute change

### PR DESCRIPTION
**Pull request by zargham-ahmad**
This PR includes the following:

- added sample_names attribute in the ramclustr object #28 
- replaced sample.names from phenoData with sample_names attribute in rc.feature files and in workflow related files ([Refactor main ramclust.R function#29](https://github.com/cbroeckl/RAMClustR/issues/29)) #29 
- Fix test cases after introducing sample_name attribute #30 
- updated namespace
- updated rc.get.csv.data to only assign MSMS data if present
- updated rc.get.csv.data and rc.get.xcms.data to use create_ramclustObj function
- formatted rc.get.xcms.data

closes #28 closes #29 closes #30

### CODE REVIEW

File: R/rc.remove.qc.R [hunk 1]
```diff
@@ -44,9 +44,9 @@ rc.remove.qc <- function(ramclustObj = NULL,
   }
 
   do.sets.rows <- sapply(
-    c(do.sets, "phenoData"),
+    c(do.sets, "phenoData", "sample_names"),
     FUN = function(x) {
-      nrow(ramclustObj[[x]])
+      NROW(ramclustObj[[x]])
```
[the remark this answers is not in the document]
Comment (zargham-ahmad) [reply R/rc.remove.qc.R:49]: small nrow doesn't works on vectors but capital NROW does treating a vector as 1-column matrix

File: NAMESPACE [hunk 2]
```diff
@@ -40,6 +41,9 @@ export(import.sirius)
 export(manual.annotation.template)
 export(mean_signal_intensities)
 export(mergeRCobjects)
+export(normalized_data_batch_qc)
+export(normalized_data_tic)
+export(order_datasets)
```
Comment (hechth) [inline NAMESPACE:46]: Not all functions need to be exported if they are not expected to be used by a user directly - maybe a function like this is more internal and doesn't have to be exported (or marked with the `@export` mark.

File: R/rc.get.csv.data.R [hunk 3]
```diff
@@ -120,17 +118,20 @@ rc.get.csv.data <- function(csv = NULL,
...
+        dimnames(data2)[[2]] <- names(MSMSdata[, featcol])
 
+        if (!all(dimnames(data1)[[2]] == dimnames(data2)[[2]])) {
+            stop("the feature names of your MS and idMSMS data are not identical")
+        }
+
+        if (!all(dimnames(data1)[[1]] == dimnames(data2)[[1]])) {
+            stop("the order and names of your MS and idMSMS data sample names are not identical")
+        }
+    }
```
Comment (hechth) [inline R/rc.get.csv.data.R:134]: Can we replace `dimnames` with the use of `rownames` and `colnames`? That would maybe make things easier.

File: R/rc.remove.qc.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (hechth) [reply R/rc.remove.qc.R:49]: This is pretty crazy.

**Review by hechth — APPROVED** [review]

File: R/rc.remove.qc.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (xtrojak) [reply R/rc.remove.qc.R:49]: well okay then :laughing:

File: R/rc.get.csv.data.R (hunk 3 above)
Comment (zargham-ahmad) [reply R/rc.get.csv.data.R]: resolved in [17f62d0](https://github.com/RECETOX/RAMClustR/pull/33/commits/17f62d0489359fc77c87e5d8a7d719e355a0eca3)

File: NAMESPACE (hunk 2 above)
Comment (zargham-ahmad) [reply NAMESPACE]: resolved in 18ad772

**Comment by hechth** [comment]
@zargham-ahmad looks good to me :) Can you add the changes to the ChangeLog? Then we can merge this :).

**Pull request merged**